src/test/app/XChain_test.cpp

```diff
@@ -0,0 +1,5120 @@
+            .tx(jtx::signers(doorA, quorum, signers))
+            .close();
+
+        scEnv.tx(bridge_create(doorB, jvb, reward, optAccountCreate))
+            .tx(jtx::signers(doorB, quorum, signers))
+            .close();
+    }
+};
+
+struct XChain_test : public beast::unit_test::suite,
+                     public jtx::XChainBridgeObjects
+{
+    XRPAmount
+    reserve(std::uint32_t count)
+    {
+        return XEnv(*this).env_.current()->fees().accountReserve(count);
+    }
+
+    XRPAmount
+    txFee()
+    {
+        return XEnv(*this).env_.current()->fees().base;
+    }
+
+    void
+    testXChainBridgeExtraFields()
+    {
+        auto jBridge = create_bridge(mcDoor)[sfXChainBridge.jsonName];
+        bool exceptionPresent = false;
+        try
+        {
+            exceptionPresent = false;
+            [[maybe_unused]] STXChainBridge testBridge1(jBridge);
+        }
+        catch (std::exception& ec)
+        {
+            exceptionPresent = true;
+        }
+
+        BEAST_EXPECT(!exceptionPresent);
+
+        try
+        {
+            exceptionPresent = false;
+            jBridge["Extra"] = 1;
+            [[maybe_unused]] STXChainBridge testBridge2(jBridge);
+        }
+        catch ([[maybe_unused]] std::exception& ec)
+        {
+            exceptionPresent = true;
+        }
+
+        BEAST_EXPECT(exceptionPresent);
+    }
+
+    void
+    testXChainCreateBridge()
+    {
+        XRPAmount res1 = reserve(1);
+
+        using namespace jtx;
+        testcase("Create Bridge");
+
+        // Normal create_bridge => should succeed
+        XEnv(*this).tx(create_bridge(mcDoor)).close();
+
+        // Bridge not owned by one of the door account.
+        XEnv(*this).tx(
+            create_bridge(mcBob), ter(temXCHAIN_BRIDGE_NONDOOR_OWNER));
+
+        // Create twice on the same account
+        XEnv(*this)
+            .tx(create_bridge(mcDoor))
+            .close()
+            .tx(create_bridge(mcDoor), ter(tecDUPLICATE));
+
+        // Create USD bridge Alice -> Bob ... should succeed
+        XEnv(*this).tx(
+            create_bridge(
+                mcAlice, bridge(mcAlice, mcGw["USD"], mcBob, mcBob["USD"])),
+            ter(tesSUCCESS));
+
+        // Create USD bridge, Alice is both the locking door and locking issue,
+        // ... should fail.
+        XEnv(*this).tx(
+            create_bridge(
+                mcAlice, bridge(mcAlice, mcAlice["USD"], mcBob, mcBob["USD"])),
+            ter(temXCHAIN_BRIDGE_BAD_ISSUES));
+
+        // Create where both door accounts are on the same chain. The second
+        // bridge create should fail.
+        XEnv(*this)
+            .tx(create_bridge(
+                mcAlice, bridge(mcAlice, mcGw["USD"], mcBob, mcBob["USD"])))
+            .close()
+            .tx(create_bridge(
+                    mcBob, bridge(mcAlice, mcGw["USD"], mcBob, mcBob["USD"])),
+                ter(tecDUPLICATE));
```

Comment:
With the most recent changes this test now returns `tesSUCCESS`, not `tecDUPLICATE`.  I read that as meaning that two different accounts on the same chain can both hold bridges for the same currencies.  That seems like a bad idea.

Am I misunderstanding the outcome of the test?

Reply:
You can now have two different account with the same bridge (although they will always be for different sides of the bridge). This is a silly thing to do if they are actually for the same chain, but we should allow this if they are for different chains, and we can't tell.

Reply:
Yup, I see your point that the bridges may be leading off to different chains.  That's fair.